rudder-web/src/main/scala/com/normation/rudder/web/snippet/administration/PropertiesManagement.scala

```diff
@@ -469,6 +470,45 @@ class PropertiesManagement extends DispatchSnippet with Loggable {
     ) apply (xml ++ Script(Run("correctButtons();") & check()))
   }
 
+  def networkProtocolSection = { xml : NodeSeq =>
+    //  initial values, updated on successfull submit
+    var initReportsProtocol = configService.report_protocol
+    var reportProtocol = initReportsProtocol.getOrElse("tcp")
+    def check() = {
+      S.notice("updateNetworkProtocol","")
+      Run(s"""$$("#networkProtocolSubmit").button( "option", "disabled",${initReportsProtocol.map(_ == reportProtocol).getOrElse(false)});""")
+    }
+
+    def submit = {
+      configService.set_report_protocol(reportProtocol).foreach(updateOk => initReportsProtocol = Full(reportProtocol))
+
+      // start a promise generation, Since we check if there is change to save, if we got there it mean that we need to redeploy
+      startNewPolicyGeneration
+      S.notice("updateNetworkProtocol","Network protocol options correctly updated")
+      check()
+    }
+
+     ( "#reportProtocol" #> {
+      initReportsProtocol match {
+        case Full(value) =>
+          SHtml.ajaxCheckbox(
+              value == "udp"
+            , (b : Boolean) => { reportProtocol = if (b) "udp" else "tcp"; check() }
+            , ("id","reportProtocol")
+          )
+          case eb: EmptyBox =>
+            val fail = eb ?~ "there was an error while fetching value of property: 'Report protocol' "
+            <div class="error">{fail.msg}</div>
```

Comment:
I think that in that case, we want to set back a correct value and present the checkbox, else the user can't converge to a correct state ever.